deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -5,371 +5,156 @@
+        try (InputStream inputStream = new ByteArrayInputStream(trustedCertificates.getBytes()))
         {
-            trustKeyStore.setCertificateEntry(TRUSTED_IOT_HUB_CERT_PREFIX + UUID.randomUUID(), c);
+            return getSSLContextFromStream(inputStream);
         }
-
-        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-        trustManagerFactory.init(trustKeyStore);
-
-        return trustManagerFactory;
     }
 
-    private static char[] generateTemporaryPassword()
+    /**
+     * Create an IotHubSSLContext that trusts the PEM formatted certificates stored in a file with the provided path.
+     * @param trustedCertificatesFilePath the absolute file path of the file that contains the PEM formatted
+     * certificates that this IotHubSSLContext will trust.
+     * @return the created IotHubSSLContext.
+     * @throws CertificateException if the provided trustedCertificates are not PEM formatted and cannot be parsed.
+     * @throws IOException if the provided trustedCertificates cannot be read as a stream.
+     * @throws KeyStoreException if a key store cannot be created.
+     * @throws NoSuchAlgorithmException if your device cannot use x509 certificates or TLS SSLContexts.
+     * @throws KeyManagementException if the created SSLContext cannot be initialized.
+     */
+    public static SSLContext getSSLContextFromFile(String trustedCertificatesFilePath) throws CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException
     {
-        char[] randomChars = new char[256];
-        SecureRandom secureRandom = new SecureRandom();
-
-        for (int i = 0; i < 256; i++)
+        if (trustedCertificatesFilePath == null || trustedCertificatesFilePath.isEmpty())
         {
-            // character will be between 97 and 122 on the ASCII table. This forces it to be a lower case character.
-            // that ensures that the password, as a whole, is alphanumeric
-            randomChars[i] = (char) (97 + secureRandom.nextInt(26));
+            throw new IllegalArgumentException("The provided certificate path string cannot be null or empty");
         }
 
-        return randomChars;
-    }
-
-    private static Key parsePrivateKey(String privateKeyString) throws CertificateException
-    {
-        try
-        {
-            // Codes_SRS_IOTHUBSSLCONTEXT_34_031: [This function shall return a Private Key instance created by the provided PEM formatted privateKeyString.]
-            Security.addProvider(new BouncyCastleProvider());
-            PEMParser privateKeyParser = new PEMParser(new StringReader(privateKeyString));
-            Object possiblePrivateKey = privateKeyParser.readObject();
-            return IotHubSSLContext.getPrivateKey(possiblePrivateKey);
-        }
-        catch (Exception e)
+        try (FileInputStream fis = new FileInputStream(trustedCertificatesFilePath))
```

Comment:
In Java, the pattern 

```java
try (InputStream is = new InputStream())
{
    // use the stream
}
```

is similar to a "using" block in .NET, so the input stream is closed no matter what when this block ends.